solr/core/src/java/org/apache/solr/metrics/otel/instruments/AttributedLongTimer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.metrics.otel.instruments;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongHistogram;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class records the elapsed time (in milliseconds) to a {@link LongHistogram}. Each thread
+ * measures its own timer allowing concurrent timing operations on separate threads..
+ */
+public class AttributedLongTimer extends AttributedLongHistogram {
+
+  private final ThreadLocal<Long> startTimeNanos = new ThreadLocal<>();
```

Comment:
The use of ThreadLocal concerns me; seems like an over-use of ThreadLocal.  I could be convinced otherwise, though.  Instead of conceptually changing the state of this timer (scoped to this thread, granted), couldn't we return a Timer object that the caller closes?  Note that Solr has RTimer / RTimerTree you may want to see... perhaps return an RTimer subclass, for example.